### PR TITLE
Fix swapping generic assay profiles in comparison view

### PR DESCRIPTION
**Pull request by MatthijsPon**
Fix https://github.com/cBioPortal/cbioportal/issues/10520

Describe changes proposed in this pull request:
- [A previous PR](https://github.com/cBioPortal/cbioportal-frontend/pull/4686) forgot to change some variable names to the new naming, this fixes the leftovers

Compare swapping data sets on the following pages:
- [Current behaviour](https://www.cbioportal.org/results/comparison?cancer_study_list=ccle_broad_2019&tab_index=tab_visualize&case_set_id=ccle_broad_2019_all&Action=Submit&gene_list=TP53&comparison_subtab=generic_assay_treatment_response)
- [Fixed behaviour (netlify)](https://deploy-preview-4922--cbioportalfrontend.netlify.app/results/comparison?cancer_study_list=ccle_broad_2019&tab_index=tab_visualize&case_set_id=ccle_broad_2019_all&Action=Submit&gene_list=TP53&comparison_subtab=generic_assay_treatment_response)

### CODE REVIEW

**Comment by MatthijsPon** [comment]
It seems the failing tests are in all PRs and is not caused by my PR? Or am I faulty?

File: src/shared/lib/comparison/ComparisonStore.ts [hunk 1]
```diff
@@ -1699,14 +1699,14 @@ export default abstract class ComparisonStore extends AnalysisStore
     readonly gaEnrichmentGroupsByAssayType = remoteData({
         await: () => [
             this
-                .selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType,
+                .selectedAllGenericAssayEnrichmentProfileMapGroupedByGenericAssayType,
```
Comment (TJMKuijpers) [inline src/shared/lib/comparison/ComparisonStore.ts:1702]: Just wondering, after your changes do we still need the following `action` and `readonly` :
- `action setGenericAssayEnrichmentProfileMap`  (line 1095)
-  `readonly selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` (line 914)

It looks like `selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` was only used in the functions where you made the changes. If they are not needed anymore, might be nice to remove them from the code (to clean up)

**Review by TJMKuijpers — COMMENTED** [review]
Might be good to check if we can remove `selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` from the source code if it is not being used anymore. (code cleanup)

File: src/shared/lib/comparison/ComparisonStore.ts (hunk 1 above)
Comment (dippindots) [reply src/shared/lib/comparison/ComparisonStore.ts:1702]: I remember that the `selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` is used by the UI to determine which profile is currently being displayed. Maybe @MatthijsPon can double-check to confirm if it is necessary or not.

**Review by dippindots — COMMENTED** [review]
It looks good to me. e2e tests are not running for some reason, we might want to see the tests passing before we merge it

File: src/shared/lib/comparison/ComparisonStore.ts (hunk 1 above)
Comment (MatthijsPon) [reply src/shared/lib/comparison/ComparisonStore.ts:1702]: @dippindots The only place where I can find `selectedGenericAssayEnrichmentProfileMapGroupedByGenericAssayType` being used, is in the file [src/pages/groupComparison/GenericAssayEnrichments.tsx](https://github.com/thehyve/cbioportal-frontend/blob/fix_ga_profile_switch/src/pages/groupComparison/GenericAssayEnrichments.tsx). This file seems to have been made obsolete by [src/pages/groupComparison/GenericAssayEnrichmentCollections.tsx](https://github.com/thehyve/cbioportal-frontend/blob/fix_ga_profile_switch/src/pages/groupComparison/GenericAssayEnrichmentCollections.tsx)?

**Review by dippindots — APPROVED** [review]
Thanks for cleaning up the code, LGTM

**Comment by inodb** [comment]
Thanks @MatthijsPon and @TJMKuijpers !

**Pull request merged**